api/tests/routes/public/individual_offers/v1/get_products_test.py

```diff
@@ -32,18 +35,52 @@ def test_should_raise_404_because_venue_provider_is_inactive(self, client):
             response = client.with_explicit_token(plain_api_key).get("%s?venueId=%s" % (self.endpoint_url, venue_id))
             assert response.status_code == 404
 
-    def test_get_first_page(self, client):
+    def test_get_first_page(self, client: TestClient):
         plain_api_key, venue_provider = self.setup_active_venue_provider()
         offers = offers_factories.ThingOfferFactory.create_batch(12, venue=venue_provider.venue)
 
         venue_id = venue_provider.venueId
         with testing.assert_num_queries(self.num_queries_success):
             response = client.with_explicit_token(plain_api_key).get(
-                f"/public/offers/v1/products?venueId={venue_id}&limit=5"
+                self.endpoint_url, params={"venueId": venue_id, "limit": 5}
             )
+
             assert response.status_code == 200
+            assert [product["id"] for product in response.json["products"]] == [offer.id for offer in offers[0:5]]
+
+    def test_should_return_all_offers(self, client):
+        plain_api_key, venue_provider = self.setup_active_venue_provider()
+        venue2 = self.setup_venue()
+        providers_factories.VenueProviderFactory(venue=venue2, provider=venue_provider.provider)
+        offers = offers_factories.ThingOfferFactory.create_batch(
+            12, venue=venue_provider.venue
+        ) + offers_factories.ThingOfferFactory.create_batch(12, venue=venue2)
+        offers_factories.ThingOfferFactory()
 
-        assert [product["id"] for product in response.json["products"]] == [offer.id for offer in offers[0:5]]
+        no_check_on_venue_num_queries = self.num_queries_success - 1
+        with testing.assert_num_queries(no_check_on_venue_num_queries):
+            response = client.with_explicit_token(plain_api_key).get(self.endpoint_url)
+
+            assert response.status_code == 200
+            assert [product["id"] for product in response.json["products"]] == [offer.id for offer in offers]
```

Comment:
(détail) il y a toujours un petit risque avec les comparaisons de listes comme ça, surtout que l'ordre n'est pas très important ici et qu'il pourrait changer à l'avenir.


```suggestion
            assert {product["id"] for product in response.json["products"]} == {offer.id for offer in offers}
```

Reply:
Alors si justement parce que ce endpoint renvoie les produits par ordre croissant d'id (c'est ce qui permet la pagination par index).

Reply:
on n'a rien vu, on n'a rien vu 😶‍🌫️